desc/grid.py

```diff
@@ -51,30 +56,47 @@ class Grid(IOAble):
         "_inverse_zeta_idx",
     ]
 
-    def __init__(self, nodes, sort=True):
+    def __init__(self, nodes, sort=True, jitable=False):
         # Python 3.3 (PEP 412) introduced key-sharing dictionaries.
         # This change measurably reduces memory usage of objects that
         # define all attributes in their __init__ method.
         self._NFP = 1
         self._sym = False
         self._node_pattern = "custom"
         self._nodes, self._spacing = self._create_nodes(nodes)
-        self._enforce_symmetry()
         if sort:
             self._sort_nodes()
-        self._axis = self._find_axis()
-        (
-            self._unique_rho_idx,
-            self._inverse_rho_idx,
-            self._unique_theta_idx,
-            self._inverse_theta_idx,
-            self._unique_zeta_idx,
-            self._inverse_zeta_idx,
-        ) = self._find_unique_inverse_nodes()
+        if jitable:
+            # dont do anything with symmetry since that changes # of nodes
+            # avoid point at the axis, for now. FIXME: make axis boolean mask?
```

Comment:
The limits in the compute function are jitable because they rely on the `axis.size` parameter which I think is a compile time constant. If you switch this to boolean mask you may need to add a `has_axis` boolean because `jnp.any(axis_mask)` may make jit complain.

Reply:
Yeah I still don't have a good idea of how to do dynamic grids that may or may not have a point on the axis, so for now just fudging it off the axis which should be fine for 99.999% of cases.